radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/epoch/EpochManager.java

```diff
@@ -312,63 +310,69 @@ private void processEpochChange(EpochChange epochChange) {
       }
     }
 
+    final var queuedEventsForNewEpoch = queuedEventsForNextEpoch;
+    queuedEventsForNextEpoch = new ArrayList<>(256);
+
     this.currentEpoch = epochChange;
     this.updateEpochState();
     this.bftEventProcessor.start();
 
-    // Execute any queued up consensus events
-    final List<ConsensusEvent> queuedEventsForEpoch =
-        queuedEvents.getOrDefault(epochChange.getEpoch(), Collections.emptyList());
-    var highView =
-        queuedEventsForEpoch.stream()
-            .map(ConsensusEvent::getView)
-            .max(Comparator.naturalOrder())
-            .orElse(View.genesis());
-
-    queuedEventsForEpoch.stream()
-        .filter(e -> e.getView().equals(highView))
-        .forEach(this::processConsensusEventInternal);
-
-    queuedEvents.remove(epochChange.getEpoch());
+    this.processCachedConsensusEventsAtStartOfEpoch(epochChange, queuedEventsForNewEpoch);
   }
 
-  private void processConsensusEventInternal(ConsensusEvent consensusEvent) {
-    this.counters.increment(CounterType.BFT_EVENTS_RECEIVED);
+  public void processConsensusEvent(ConsensusEvent consensusEvent) {
+    final var currentEpoch = this.currentEpoch();
```

Comment:
I believe we can just use `this.currentEpoch()` instead of creating a variable.

Reply:
This was a remnant from where I was concerned with race conditions - but I think this is still fine - as it means someone reading the method doesn't have to worry about `this.currentEpoch()` changing mid-method, which decreases the intellectual burden. (of course, if you know that all of this runs in a single thread, I guess this is irrelevant)

Reply:
Sonar says it is a major code smell. If it wasn't single-threaded, I'm not sure we should change this field arbitrarily.

Reply:
Seems Sonar is complaining cause both variables have the same name.